From 3f8ee50aef4120313ab91a9aa3dd1439b74064dc Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 18 Nov 2014 14:20:55 +0800
Subject: [PATCH 0325/5242] MLK-9866: mfd: si476x: FM will fail to open
 sometimes.

commit  ccb87757949396f17aeb14b24657b0777860b308 from
https://source.codeaurora.org/external/imx/linux-imx.git

In commit e856a0ebc23dcd2c933e3f902317652cc50f0067, we disabled
wait_event_timeout for CMD_POWER_DOWN, which will cause power down
failed sometimes, then FM will fail to reopen.
In this patch enable the wait_event_timeout for power down.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 30aa7f1a3580d0a5bc866c624d1da441fd5502c1)
(cherry picked from commit 2b1990fee264d805fcdc0005dcf8b5b63b261d51)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mfd/si476x-cmd.c |   14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/mfd/si476x-cmd.c b/drivers/mfd/si476x-cmd.c
index d5cfe7b..2086b46 100644
--- a/drivers/mfd/si476x-cmd.c
+++ b/drivers/mfd/si476x-cmd.c
@@ -303,13 +303,13 @@ static int si476x_core_send_command(struct si476x_core *core,
 	 * possible racing conditions when working in polling mode */
 	atomic_set(&core->cts, 0);
 
-	if (!(command == CMD_POWER_DOWN))
-		if (!wait_event_timeout(core->command,
-					atomic_read(&core->cts),
-					usecs_to_jiffies(usecs) + 1))
-			dev_warn(&core->client->dev,
-				"(%s) [CMD 0x%02x] Answer timeout.\n",
-				__func__, command);
+	/* if (unlikely(command == CMD_POWER_DOWN) */
+	if (!wait_event_timeout(core->command,
+				atomic_read(&core->cts),
+				usecs_to_jiffies(usecs) + 1))
+		dev_warn(&core->client->dev,
+			 "(%s) [CMD 0x%02x] Answer timeout.\n",
+			 __func__, command);
 
 	/*
 	  When working in polling mode, for some reason the tuner will
-- 
1.7.9.5

